core/src/edu/rice/rbox/Replication/ReplicaManagerGrpc.java

```diff
@@ -156,17 +156,46 @@ public void handleUnsubscribe(Rbox.UnsubscribeRequest request, StreamObserver<Em
 
         @Override
         public void handleUpdate(Rbox.UpdateMessage request, StreamObserver<Empty> responseObserver) {
-            // TODO: Call ChangeReceiver
+            logger.log(Level.INFO, "Handling update...");
+
+            // Call ChangeReceiver
+            // TODO: Cannot reference receiveChange from static context?
+            ChangeReceiver.receiveChange(request.getRemoteChange());
```

Comment:
Should do `changeReceiver.receiveChange` instead?

Reply:
And if we're updating the secondary, we should also notify the registrar about it.